### DISCUSSION TITLE
Can I use oxc for my project?

### DISCUSSION THREAD

**Discussion by olrtg** [oxc-project/oxc#6247] [Q&A] [answered]
Hey! I been wanting to create a language server that gives code actions for js/ts in general and frontend in frameworks/libs. Mostly inspired by https://github.com/wix-incubator/vscode-glean, https://github.com/xsburg/vscode-javascript-booster and other code transformations from products like the IDEs from Jetbrains.

Now my question is, can I use oxc to parse the file and transform the code (something like https://github.com/facebook/jscodeshift can do, parse to an AST, modify the AST and apply those modifications)? I see that there is a package for parsing and transforming the code but from the APIs and features mentioned in the docs is not clear to me if that's possible.

If you need more context just let me know, I'm very new to this stuff, sorry if it's not clear from the beginning.

**Comment by Boshen** [comment, marked as answer]
In Rust yes by using crates. In JavaScript no by using npm package.

**Reply by olrtg** [reply]
Hello! thank you so much for the quick reply 🤘🏽

Ok I see.. and in terms of performance, I now that oxc in general is probably the fastest, but, would you say in a language server built with node and using jscodeshift, do you think that there would be a noticeable difference in terms of performance?

I mean, I would like to build something performant and with great features, but also learning Rust will be quite the endeavor for a side project, but not closed to the idea if necessary. Do you have any thoughts? 🤔

**Reply by Boshen** [reply]
My suggestion is to build a few prototypes with different tool choices.

**Reply by olrtg** [reply]
Ok! I appreciate the recommendation. Thanks again 🤘🏽 good luck.
